### PR TITLE
Add tests for ChannelBase<T>

### PR DESCRIPTION
**Pull request by roncain**
This class is part of ClientBase<T> and provides a mechanism
for custom channels that do not use dynamic proxy generation.

### CODE REVIEW

**Comment by roncain** [comment]
Reviewers: code coverage analysis of ClientBase<T> revealed (at least to me) a completely untested extensibility point -- namely using a custom ChannelBase<T> inside a custom ClientBase<T>.  This extensibility point allows developers to bypass automatic proxy generation and instead to supply their own dispatching proxy for the channel.  Usage data shows it is used in Silverlight, Phone Store, and Astoria.xamarin apps, so it is reasonable to assume some apps may rely on it.

No product changes were needed for this PR, but at least these new tests demonstrate this extensibility point works.

File: src/System.Private.ServiceModel/tests/Scenarios/Client/ClientBase/ChannelBaseTests.4.0.0.cs [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+            string actual = String.Join(",", eventsCalled);
+
+            Assert.True(String.Equals(expected, actual),
+                        String.Format("Expected client to receive events '{0}' but actual was '{1}'", expected, actual));
+
+            actual = String.Join(",", proxyEventsCalled);
+            Assert.True(String.Equals(expected, actual),
+                        String.Format("Expected proxy to receive events '{0}' but actual was '{1}'", expected, actual));
+
+            // *** CLEANUP *** \\
```
Comment (StephenBonikowsky) [inline src/System.Private.ServiceModel/tests/Scenarios/Client/ClientBase/ChannelBaseTests.4.0.0.cs]: Maybe add a simple comment explaining that the closing of both proxies is supposed to have already happened and been verified above.

File: src/System.Private.ServiceModel/tests/Scenarios/Client/ClientBase/ChannelBaseTests.4.0.0.cs [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
```
Comment (StephenBonikowsky) [inline src/System.Private.ServiceModel/tests/Scenarios/Client/ClientBase/ChannelBaseTests.4.0.0.cs]: Same as with prior test.

File: src/System.Private.ServiceModel/tests/Scenarios/Client/ClientBase/ChannelBaseTests.4.0.0.cs [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+            string actual = String.Join(",", eventsCalled);
+
+            Assert.True(eventsCalled.Count == 0,
+                        String.Format("Expected client NOT to receive events but actual was '{0}'", actual));
+
+            actual = String.Join(",", proxyEventsCalled);
+            Assert.True(proxyEventsCalled.Count == 0,
+                        String.Format("Expected proxy NOT to receive events but actual was '{0}'", actual));
+
+            // *** CLEANUP *** \\
```
Comment (StephenBonikowsky) [inline src/System.Private.ServiceModel/tests/Scenarios/Client/ClientBase/ChannelBaseTests.4.0.0.cs]: Should whatever is expected to not have been closed be closed here?

**Comment by StephenBonikowsky** [comment]
LGTM

**Pull request merged**